src/main/java/org/broadinstitute/hellbender/engine/FeatureManager.java

```diff
@@ -42,34 +45,33 @@
     private static final Logger logger = LogManager.getLogger(FeatureManager.class);
 
     /**
-     * We will search these packages at startup to look for FeatureCodecs
-     */
-    private static final List<String> CODEC_PACKAGES = Arrays.asList("htsjdk.variant",
-                                                                     "htsjdk.tribble",
-                                                                     "org.broadinstitute.hellbender.utils.codecs");
-
-    /**
      * All codecs descend from this class
      */
     private static final Class<FeatureCodec> CODEC_BASE_CLASS = FeatureCodec.class;
 
     /**
      * The codec classes we locate when searching CODEC_PACKAGES
      */
-    private static final Set<Class<?>> DISCOVERED_CODECS;
+    private static Set<Class<?>> DISCOVERED_CODECS = null;
 
     /**
      * Feature arguments in tools are of this type
      */
     private static final Class<FeatureInput> FEATURE_ARGUMENT_CLASS = FeatureInput.class;
 
     /**
-     * At startup, walk through the packages in CODEC_PACKAGES, and save any (concrete) FeatureCodecs discovered
-     * in DISCOVERED_CODECS
+     * Walk through the packages in {@link GATKConf#getCodecPackages()}, and save any (concrete) FeatureCodecs discovered
+     * in DISCOVERED_CODECS.
+     *
+     * @throws IllegalStateException if codecs were alredy initialized.
      */
-    static {
+    public synchronized static void setConfiguration(final GATKConf configuration) {
```

Comment:
Replace with `setCodecPackages()`, called from `CommandLineProgram` using the information from the master configuration object.